src/partitioning/block.jl

```diff
@@ -16,7 +16,7 @@ BlockPartition(sides::NTuple{Dim,T}) where {Dim,T} = BlockPartition{Dim,T}(sides
 
 BlockPartition(sides::Vararg{T,Dim}) where {Dim,T} = BlockPartition(sides)
 
-function partition(object, method::BlockPartition)
+function partition(object, method::BlockPartition, calculate_metadata = false)
```

Comment:
The option should be part of the `BlockPartition` struct. And you should pass it to the constructor of the struct as a keyword argument `neighbors=false` by default.

Reply:
Thank you for the review. Working on it :+1: